src/empty.js

```diff
@@ -1,21 +1,38 @@
 var _curry1 = require('./internal/_curry1');
-var _hasMethod = require('./internal/_hasMethod');
 
 
 /**
- * `empty` returns an empty list for any argument, except when the argument satisfies the
- * Fantasy-land Monoid spec. In that case, this function will return the result of invoking
- * `empty` on that Monoid.
+ * Returns the empty value of its argument's type. The argument must be a
+ * [Monoid](https://github.com/fantasyland/fantasy-land#monoid), Array,
```

Comment:
Although this is not newly introduced, perhaps we should clean it up while we're here.  There is nothing here that I see that requires this to be a Monoid.  Monoids are a subset of Semigroups.  But there is nothing here that requires a closed associative binary operation on the set, the definition of Semigroup, `concat` in FantasyLand terms.

All the implementation needs is to have an `empty` function returning an element of its type.  Is there any reason to document it as actually then needing a full Monoid?